src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -671,6 +677,33 @@ void mutate(const gsl::not_null<DataBox<TagList>*> box, Invokable&& invokable,
                                               tmpl::pin<full_mutated_items>,
                                               tmpl::get_source<tmpl::_1>>>,
                       tmpl::get_destination<tmpl::_1>>>;
+  if constexpr (not std::is_same_v<
+                    decltype(invokable(
```

Comment:
can you just use `std::invoke_result` here?

Reply:
Unfortunately, not directly because `std::invoke_result` doesn't define the `type` alias when the result is void. I tried that, and found it pretty surprising that it didn't act the same as the decltype.